apps/minifront/src/state/status.ts

```diff
@@ -1,36 +1,37 @@
-import { SliceCreator } from '.';
-import { viewClient } from '../clients';
+import { ZQueryState } from '@penumbra-zone/zquery/src/types';
+import { SliceCreator, useStore } from '.';
+import { createZQuery } from '@penumbra-zone/zquery';
+import { getStatusStream } from '../fetchers/status';
 
-export interface StatusSlice {
-  start: () => Promise<void>;
+interface Status {
   fullSyncHeight?: bigint;
   latestKnownBlockHeight?: bigint;
-  error?: unknown;
 }
 
-export const createStatusSlice = (): SliceCreator<StatusSlice> => set => ({
-  start: async () => {
-    try {
-      // statusStream sends new data to stream only when a new block is detected.
-      // This can take up to 5 seconds (time of new block generated).
-      // Therefore, we need to do a unary request to start us off.
-      const status = await viewClient.status({});
-      set(state => {
-        state.status.fullSyncHeight = status.fullSyncHeight;
-        if (!status.catchingUp) state.status.latestKnownBlockHeight = status.fullSyncHeight;
-      });
-
-      const stream = viewClient.statusStream({});
-      for await (const result of stream) {
-        set(state => {
-          state.status.fullSyncHeight = result.fullSyncHeight;
-          state.status.latestKnownBlockHeight = result.latestKnownBlockHeight;
-        });
-      }
-    } catch (error) {
-      set(state => {
-        state.status.error = error;
-      });
-    }
+export const { status, useStatus } = createZQuery({
+  name: 'status',
+  fetch: getStatusStream,
+  stream: (_prevState: Status | undefined, item: Status): Status => ({
+    fullSyncHeight: item.fullSyncHeight,
+    latestKnownBlockHeight: item.latestKnownBlockHeight,
+  }),
+  getUseStore: () => useStore,
+  get: state => state.status.status,
+  set: setter => {
+    const newState = setter(useStore.getState().status.status);
+    useStore.setState(state => {
+      state.status.status = newState;
+    });
   },
 });
+
+export interface StatusSlice {
+  status: ZQueryState<{
+    fullSyncHeight?: bigint;
+    latestKnownBlockHeight?: bigint;
+  }>;
+}
+
+export const createStatusSlice = (): SliceCreator<StatusSlice> => () => ({
```

Comment:
[thought](https://conventionalcomments.org/): As a user, I'd expect to use the state inside Zustand with selectors as we always do, so I'd like for ZQuery to just return a slice that I can simply plug into the store and play with it. I don't think the return from ZQuery `{ status, useStatus }` is really relevant. Can we make ZQuery just a kinda create-and-forget type, so all the interactions with it would be through Zustand?

Reply:
That was actually my initial design for it, but one of the goals of this project was to integrate into the React lifecycle hooks so that data would get fetched on component mount, and any in-flight requests would be aborted on component unmount. Using just selectors means that we would miss out on that benefit.

I do understand the desire for something that "feels" more like Zustand, though. If we decide to merge this PR as-is, we could still revisit that question in the future.